datasetinsights/io/config_handler.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import re
+import tempfile
+from pathlib import Path
+
+from yacs.config import CfgNode
+
+import datasetinsights.constants as const
+from datasetinsights.io import create_downloader
+from datasetinsights.io.exceptions import InvalidOverrideKey
+
+logger = logging.getLogger(__name__)
+_EQUAL_STR = "="
+_REGEX_PTRN = r".*?=.*"
+
+
+def prepare_config(path=None, override=None):
+    """load and override config from local or remote locations .
+
+    Args:
+        path: config location
+        override: key-value pairs to override config
+    Returns:
+        config object of type yacs.config.CfgNode
+    """
+    config = load_config(path)
+    logger.info(f"config loading from {path} completed")
+    if override:
+        logger.info(f"overriding config params= {override}")
+        override_config(override=override, config=config)
+    return config
+
+
+def load_config(path):
+    """ load config from local or remote locations .
+    Examples:
+        >>> path="gs://thea-dev/../config.yaml"
+        >>> path="https://thea-dev/../config.yaml"
+        >>> path="http://thea-dev/../config.yaml"
+        >>> path="file:///root/../config.yaml" # absolute path
+        >>> path="file://datasetinsights/../config.yaml" # relative local path
+
+    Returns:
+        config object of type yacs.config.CfgNode
+    """
+    if not path.startswith(const.LOCAL_FILE_BASE_STR):
+        try:
+            downloader = create_downloader(source_uri=path)
+            with tempfile.TemporaryDirectory() as tmp:
+                downloader.download(source_uri=path, output=tmp)
+                logger.debug(f"downloading to directory: {tmp}")
+                config_path = Path(path)
+                file_path = tmp / config_path.relative_to(config_path.parent)
+                return CfgNode.load_cfg(open(file_path, "r"))
+        except Exception as e:
+            logger.exception(f"Config loading from location{path} failed")
```

Comment:
If the users use `s3://...` as the config path, they will see this error message, but it's not clear why this failed. I think we should validate against supported config path prefix, as in https://github.com/Unity-Technologies/datasetinsights/blob/master/datasetinsights/commands/download.py#L12